### PR TITLE
Add minor performance improvements to resnet input pipeline

### PR DESCRIPTION
**Pull request by guptapriya**
Added the following changes to improve input pipeline:
1. Remove usage of one_hot labels  (~2% improvement)
2. Add drop_remainder to batching. This helps in input shape being determined ahead of time which helps improve performance. (~2% improvement)
3. Use parallel_interleave for imagenet dataset reading. (~3% improvement)

The improvements were observed when testing with 8 V100 GPUs on DGX-1 using mirrored strategy (fp16, resnet v2). The differences may not be noticeable in other cases where input pipeline is not the bottleneck.

### CODE REVIEW

File: official/resnet/resnet_run_loop.py [hunk 1]
```diff
@@ -79,7 +79,8 @@ def process_record_dataset(dataset, is_training, batch_size, shuffle_buffer,
       tf.contrib.data.map_and_batch(
           lambda value: parse_record_fn(value, is_training),
           batch_size=batch_size,
-          num_parallel_batches=1))
+          num_parallel_batches=1,
+          drop_remainder=True))
```
[the remark this answers is not in the document]
Comment (karmel) [reply official/resnet/resnet_run_loop.py]: It should not matter much as long as the batch size is reasonable, since this only drops the part of the dataset that doesn't fit into a full batch.

File: official/resnet/imagenet_main.py [hunk 2]
```diff
@@ -177,6 +175,10 @@ def input_fn(is_training, data_dir, batch_size, num_epochs=1):
     dataset = dataset.shuffle(buffer_size=_NUM_TRAIN_FILES)
 
   # Convert to individual records
+  # TODO(guptapriya): Should we make this cycle_length a flag similar to
```
[the remark this answers is not in the document]
Comment (guptapriya) [reply official/resnet/imagenet_main.py]: No I haven't had the chance to run on things other than DGX-1V. I don't think the performance difference will show up on K80s because the input pipeline will not be the bottleneck. But I haven't tested it. I am talking to the input team to figure out if a constant here makes sense, or should this be tuned (in which case we may need to just remove it)

File: official/resnet/resnet_run_loop.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (guptapriya) [reply official/resnet/resnet_run_loop.py]: Right, it only drops the last partial batch. so for 60k images for a batch size of 2048 also it should drop only 608 images or so.

**Comment by pkulzc** [comment]
The object detection change in this PR has already been approved and merged in other PR so object detection reviews are not necessary. I'm removing derekjchow, jch1 and me from reviewers.
Please let me know if further reviews from us are required. Thanks!

**Comment by guptapriya** [comment]
@pkulzc  - yes! I have messed up this PR while doing rebase etc.. Hence it pulled in some already merged changes :( Sorry about the spam. I will fix it, and yes, no action is needed from you guys!
